cli/http_util.rs

```diff
@@ -26,6 +30,25 @@ pub fn get_client() -> Client<Connector, hyper::Body> {
   Client::builder().build(c)
 }
 
+/// Construct the next uri based on base uri and location header fragment
+/// See <https://tools.ietf.org/html/rfc3986#section-4.2>
+fn resolve_url_from_location(base_url: &Url, location: &str) -> Url {
+  if location.starts_with("http://") || location.starts_with("https://") {
+    // absolute uri
+    Url::parse(location).expect("provided redirect url should be a valid url")
+  } else if location.starts_with("//") {
+    // "//" authority path-abempty
+    Url::parse(&format!("{}:{}", base_url.scheme(), location))
+      .expect("provided redirect url should be a valid url")
+  } else if location.starts_with('/') {
+    // path-absolute
+    base_url.join(location).unwrap()
+  } else {
+    // assuming path-noscheme | path-empty
+    base_url.join(location).unwrap()
```

Comment:
This is probably wrong... @kevinkassimo can you explain this bit?
```rust
} else {
  // assuming path-noscheme | path-empty
  let mut new_uri_parts = base_uri.clone().into_parts();
  let base_uri_path_str = base_uri.path().to_owned();
  let segs: Vec<&str> = base_uri_path_str.rsplitn(2, '/').collect();
  new_uri_parts.path_and_query = Some(
    format!("{}/{}", segs.last().unwrap_or(&""), location)
      .parse()
      .unwrap(),
  );
  Uri::from_parts(new_uri_parts).unwrap()
}
```
I don't really understand purpose of `rsplitn`

Reply:
Alright, I think I figured it out. Pop last part of url (after last slash).